### ISSUE TITLE
Potential better results on Camelyon16

### ISSUE THREAD

**Issue by Jiawei-Yang** [binli123/dsmil-wsi#29]
Hi, dear Li, thanks for your fabulous open-sourced project!

I'm trying to reproduce your results on Camelyon16 and found better performance can be achieved.

The performance of DSMIL-LC is reported to have an accuracy of 0.8992 and an AUC of 0.9165 in Table 1 of the DSMIL paper. The posted log in issue #14 shows a higher performance than the reported results after 200-epoch training, i.e., `average score: 0.9000, AUC: class-0>>0.927734375`, and even better performance at the early training stage, e.g., `average score: 0.9250, AUC: class-0>>0.9446614583333333`. Which model is used to report the final results? the best one or the 200-epoch one?

Also, I modified some parts of the code to speed up the training (e.g., I removed the `drop_patch()` part since the drop_rate is 0, but I noticed the drop_patch is still functional when the drop_rate is 0 (which is another shuffling, right?) ). 

After modification, I trained the DSMIL model on the precomputed Camelyon16 features and got a performance of `average score: 0.9375, AUC: class-0>>0.972` after 200-epoch training. In your experience, how much variation do the results have? These stronger results would definitely improve your state-of-the-art results.

BTW, just for clarification, which split was the model evaluated on? The released code uses the former 80% of the csv file as the training set and the latter 20% of the csv file as the test set, leading to a 319/80 WSIs for training/testing. The Camelyon16 is reported to have a split of 270/129 for training/testing in the DSMIL paper.

Another BTW: would you like to share the DSMIL model weights other than the embedder?

Thanks in advance for your reply. Your work is great!

**Comment by binli123** [comment]
> Hi, dear Li, thanks for your fabulous open-sourced project!
> 
> I'm trying to reproduce your results on Camelyon16 and found better performance can be achieved.
> 
> The performance of DSMIL-LC is reported to have an accuracy of 0.8992 and an AUC of 0.9165 in Table 1 of the DSMIL paper. The posted log in issue #14 shows a higher performance than the reported results after 200-epoch training, i.e., `average score: 0.9000, AUC: class-0>>0.927734375`, and even better performance at the early training stage, e.g., `average score: 0.9250, AUC: class-0>>0.9446614583333333`. Which model is used to report the final results? the best one or the 200-epoch one?
> 
> Also, I modified some parts of the code to speed up the training (e.g., I removed the `drop_patch()` part since the drop_rate is 0, but I noticed the drop_patch is still functional when the drop_rate is 0 (which is another shuffling, right?) ).
> 
> After modification, I trained the DSMIL model on the precomputed Camelyon16 features and got a performance of `average score: 0.9375, AUC: class-0>>0.972` after 200-epoch training. In your experience, how much variation do the results have? These stronger results would definitely improve your state-of-the-art results.
> 
> BTW, just for clarification, which split was the model evaluated on? The released code uses the former 80% of the csv file as the training set and the latter 20% of the csv file as the test set, leading to a 319/80 WSIs for training/testing. The Camelyon16 is reported to have a split of 270/129 for training/testing in the DSMIL paper.
> 
> Another BTW: would you like to share the DSMIL model weights other than the embedder?
> 
> Thanks in advance for your reply. Your work is great!

1. In the code random train/test split is used while in the paper the original 270/129 split is used, which should mainly be the reason why the results are better.
2. I recall that you might be able to use `python download.py --dataset=c16-test` to download the aggregator weights for Camelyon16.

**Comment by Jiawei-Yang** [comment]
Hi, Li

When I re-trained the MILNet with the 270/129 split instead of the 80%/20% split, I found the best accuracy is 0.9535, higher than 0.8992 reported in the paper. 

Besides, I used the provided `aggregator.pth` for testing. It gives an accuracy of 0.9535 as well, higher than 0.8992.

I guess you might re-train the SimCLR to provide the open-source patch embeddings, and the re-training process might use the wrong split? For example, using 80% data (319 WSIs) to pre-train SimCLR, instead of 270 WSIs. Parts of official testing slides might be exposed to the embedder's training process.

**Comment by binli123** [comment]
> Hi, Li
> 
> When I re-trained the MILNet with the 270/129 split instead of the 80%/20% split, I found the best accuracy is 0.9535, higher than 0.8992 reported in the paper.
> 
> Besides, I used the provided `aggregator.pth` for testing. It gives an accuracy of 0.9535 as well, higher than 0.8992.
> 
> I guess you might re-train the SimCLR to provide the open-source patch embeddings, and the re-training process might use the wrong split? For example, using 80% data (319 WSIs) to pre-train SimCLR, instead of 270 WSIs. Parts of official testing slides might be exposed to the embedder's training process.

This is probably due to two reasons: `aggregator.pth` is trained on an 80/20 split instead of the original split. After the paper, we have been continuously training the embedder for quite some days, there are multiple sets of weights you can find here: https://drive.google.com/drive/folders/14pSKk2rnPJiJsGK2CQJXctP7fhRJZiyn?usp=sharing.

**Comment by Jiawei-Yang** [comment]
Thanks for your quick response!

I also evaluated the provided `aggregator.pth` on the 80%/20% split, which gives an accuracy of 0.9250, and an AUC of 0.9667.

Do you think it is reasonable to stick to the 80%/20% split rather than the original split for further study or comparison?

Thank you for your time!
